src/Symfony/Component/HttpClient/NoPrivateNetworkHttpClient.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpClient\Exception\InvalidArgumentException;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpFoundation\IpUtils;
```

Comment:
we should throw a LogicException when the class is missing, to invite to install http-foundation (see similar exceptions in the source code - look for "composer require")

Reply:
I did what CachingHttpClient does, except that I removed version information from exception string, because IpUtils was introduced on version 2.2. Is that OK or should I use ^4.4?

Reply:
CachingHttpClient takes a class from HttpKernel as argument so it cannot be used without the component, that's why there is no check there.

Reply:
```php
<?php
public function __construct(HttpClientInterface $client, $subnets = null)
{
    if (!class_exists(IpUtils::class)) {
        throw new \LogicException(sprintf('Using "%s" requires that the HttpFoundation component is installed, try running "composer require symfony/http-foundation".', __CLASS__));
    }

    // ...
}```

Reply:
maybe this would be more consistent with existing errors:
`'You can not use "%s" if the HttpFoundation component is not installed. Try running "composer require symfony/http-foundation".'`